File: React Front-end/src/services/RoomService.js
import axios from 'axios'

const url1 ="http://localhost:8080/getVaccantRooms"
const url2 = "http://localhost:8080/getroom"
const url3 = "http://localhost:8080/addroom"
const url4 = "http://localhost:8080/updateroom"
const url5 ="http://localhost:8080/getoneroom"
const URL6 = "http://localhost:8080/deleteroom"

class RoomService {

    getvaccantroom(){
        return axios.get(url1);
    }

    getallroom(){
        return axios.get(url2);
    }
    addroom(rooms) {
        return axios.post(url3,rooms);
    }
    updateroom(rooms,roomId){
        return axios.put(url4+'/'+ roomId,rooms)
    }
    getRoomById(roomId){
        return axios.get(url5+'/' + roomId)
    }
    deleteRoom(roomId){
        return axios.delete(URL6+'/'+roomId);
    }
}
export default new RoomService();

File: Hotel-Management-Project/src/main/java/com/quinnox/hotelmanagementproject/service/RoomService.java
package com.quinnox.hotelmanagementproject.service;

import java.util.List;

import com.quinnox.hotelmanagementproject.dto.RoomDto;
import com.quinnox.hotelmanagementproject.entities.Room;

public interface RoomService {

	public boolean addRoom(RoomDto room);

	public List<Room> getAllRooms();

	public boolean updateRoom(int roomId,RoomDto room);

	public boolean deleteRoom(int roomId);

	public Room getOneRoom(int roomId);
	
	public List<Room> getVaccantRooms();

}
